src/WebJobs.Extensions.CosmosDB/Config/CosmosDBExtensionConfigProvider.cs

```diff
@@ -158,5 +168,49 @@ public override bool IsMatch(Type type, OpenTypeMatchContext context)
                 return base.IsMatch(type, context);
             }
         }
+
+        // Cannot use `new BinaryData(cosmosAttribute)` because this "may only be called on a Type for
+        // which Type.IsGenericParameter is true"; therefore we use our own reference type.
+        // Has the same properties as CosmosDBAttribute
+        private class CosmosParameterBindingDataContent
+        {
+            public CosmosParameterBindingDataContent()
```

Comment:
Curious, if we need this default constructor for any specific scenarios?

Reply:
I don't think there is a scenario for this other than maybe for testing? Happy to remove it :)